docs/template_plugin/tests/functional/op_reference/generate_proposals.cpp

```diff
@@ -0,0 +1,328 @@
+        const auto anchors = std::make_shared<op::v0::Parameter>(params.inType, params.anchorsShape);
+        const auto deltas = std::make_shared<op::v0::Parameter>(params.inType, params.deltasShape);
+        const auto scores = std::make_shared<op::v0::Parameter>(params.inType, params.scoresShape);
+        const auto GenerateProposal = std::make_shared<op::v9::GenerateProposals>(im_info,
+                                                                    anchors,
+                                                                    deltas,
+                                                                    scores,
+                                                                    params.attrs);
+        GenerateProposal->set_roi_num_type(params.roiNumType);
+        return std::make_shared<ov::Model>(GenerateProposal->outputs(), ParameterVector {im_info, anchors, deltas, scores});
+    }
+};
+
+TEST_P(ReferenceGPLayerTest, CompareWithRefs) {
+    Exec();
+}
+
+template <element::Type_t IN_ET, element::Type_t OUT_RT>
+std::vector<GPParams> generateGPFloatParams() {
+    using T = typename element_type_traits<IN_ET>::value_type;
+    using RT = typename element_type_traits<OUT_RT>::value_type;
+
+    Attrs attrs;
+    attrs.min_size = 1.0;
+    attrs.nms_threshold = 0.699999988079071;
+    attrs.pre_nms_count = 1000;
+    attrs.post_nms_count = 6;
+    attrs.nms_eta = 1.0;
+    attrs.normalized = true;
+
+    std::vector<GPParams> generateProposalParams {
+        GPParams(attrs,
+                              1,
+                              3,    // A
+                              2,    // H
+                              6,    // W
+                              IN_ET,
+                              OUT_RT,
+                              std::vector<T>{1.0f, 1.0f, 0.0f},
+                              std::vector<T>{1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
+                                             1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f,
```

Comment:
There is a constructor of vetcor for doing this: std::vector<T>(count, init_value)

Reply:
updated